### PR TITLE
Update to operator-sdk 1.23.0

### PR DESCRIPTION
**Pull request by littlejawa**
Following steps from https://sdk.operatorframework.io/docs/upgrading-sdk-version/v1.23.0/

Makefile:
- do not download binaries when they are found already
- update controller-tools to 0.9.2
- update kustomize to 4.5.5

go.mod:
- update controller-runtime to 0.12.2
- update kubernetes dependencies to 0.24.2
- make sure controller-tools is still there, and updated to 0.9.2

manager_auth_proxy_patch.yaml:
- update kube-rbac-proxy to 0.13.0 (using digest)

config/default/kustomization.yaml:
- changes related to updating to kustomize 4.5.5

misc:
- updated developper's doc
- modified bundle-custom.Dockerfile to use operator-sdk 1.23

### CODE REVIEW

**Comment by littlejawa** [comment]
I'm marking it "WIP" because I've not been able to test it.
When installing kataconfig, I get an error:
`Error "failed calling webhook "vkataconfig.kb.io": failed to call webhook: Post "https://controller-manager-service.openshift-sandboxed-containers-operator.svc:443/validate-kataconfiguration-openshift-io-v1-kataconfig?timeout=10s": no endpoints available for service "controller-manager-service"" for field "undefined".`

Building / installing the operator from "main" just works, so the error seems to be related to the changes in this PR.

I don't know what's wrong. Maybe it's related to the update in kustomize (there is a warning in the document about that, but I don't understand the details).
Putting it there to get other eyes on it, and get it tested on the CI (outside of my own environment) just to see if it is better.
I'll continue digging into it until I get it to work.

**Review by cpmeadors — APPROVED** [review]
All changes look good based on updated doc.  controller-gen is still in go.mod.

**Comment by jensfr** [comment]
> I'm marking it "WIP" because I've not been able to test it. When installing kataconfig, I get an error: `Error "failed calling webhook "vkataconfig.kb.io": failed to call webhook: Post "https://controller-manager-service.openshift-sandboxed-containers-operator.svc:443/validate-kataconfiguration-openshift-io-v1-kataconfig?timeout=10s": no endpoints available for service "controller-manager-service"" for field "undefined".`
> 
> Building / installing the operator from "main" just works, so the error seems to be related to the changes in this PR.
> 
> I don't know what's wrong. Maybe it's related to the update in kustomize (there is a warning in the document about that, but I don't understand the details). Putting it there to get other eyes on it, and get it tested on the CI (outside of my own environment) just to see if it is better. I'll continue digging into it until I get it to work.

We usually see that error when the controller pod is not running. what does `oc describe pod <podname> say?

File: config/default/manager_auth_proxy_patch.yaml [hunk 1]
```diff
@@ -10,7 +10,7 @@ spec:
     spec:
       containers:
       - name: kube-rbac-proxy
-        image: gcr.io/kubebuilder/kube-rbac-proxy@sha256:a4f6c39b17f2ba68366c6d4e139ed18ede2ad9f3f9d5bf7477bdcbef73b43e54
+        image: gcr.io/kubebuilder/kube-rbac-proxy@sha256:31eb553d98f6b81054babf56b952f7f2f4b56e1c782e2da4feb6ab5247bd288c
```
Comment (jensfr) [inline config/default/manager_auth_proxy_patch.yaml:13]: reminder to double check this digest

**Comment by littlejawa** [comment]
Fixed the digest for kube-rbac-proxy v0.13.0, and I could test it successfully.
Thanks for the help :)

**Review by jensfr — APPROVED** [review]
lgtm, thank you @littlejawa

**Pull request merged**